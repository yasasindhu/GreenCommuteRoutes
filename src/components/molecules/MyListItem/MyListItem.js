import React from 'react'
import { ListItem, ListItemIcon, ListItemText, ThemeProvider, Grid, makeStyles, Typography } from '@material-ui/core';
import TurnedInNotOutlinedIcon from '@material-ui/icons/TurnedInNotOutlined';
import customtheme from '../../theme';


const useStyles = makeStyles((theme) => ({
  selectedItemStyle: {
    backgroundColor: 'rgba(90, 197, 104, 0.15)',
    borderRadius: '10px',
    height: '50px',
    width: '240px',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '8px 10px',
    cursor: 'pointer',
  },
  selectedIcon: {
    color: '#5ac568',
    width: '24px',
    height: '24px',
  },
  selectedTypographyStyle: {
    fontSize: '16px',
    color: '#5ac568',
    fontFamily: 'Montserrat',
    lineHeight: '24px',
    letterSpacing: '0.1px',
  },
}));


export default function MyListItem(props) {
  const classes = useStyles();
  return (
    <ThemeProvider theme={customtheme}>
      <Grid direction="column" container className={classes.selectedItemStyle} data-testid="divinmylistitem">
        <ListItem data-testid="listiteminmylistitem">
          <ListItemIcon data-testid="listitemiconinlistitem">
            <TurnedInNotOutlinedIcon className={classes.selectedIcon} />
          </ListItemIcon>
          <ListItemText data-testid="listitemtextinlistitem" primary={<Typography variant="subtitle1" className={classes.selectedTypographyStyle}>{"Saved Jobs"}</Typography>} />
        </ListItem>
      </Grid>
    </ThemeProvider>
  );
}