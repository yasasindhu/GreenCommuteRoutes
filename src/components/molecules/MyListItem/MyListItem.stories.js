import React from "react";

import MyListItem from "./MyListItem";


export default {
    title: "molecules/SideNavbarTab",
    component: MyListItem,
};
const Template = (args) => <MyListItem {...args} />;

export const PrimaryListItem = Template.bind({});

