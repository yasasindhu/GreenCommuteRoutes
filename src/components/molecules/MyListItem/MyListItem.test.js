import { render} from '@testing-library/react';
import MyListItem from './MyListItem';



describe('Typography Rendered', () => {
    test('listitemtextinlistitem', () => {
      const { asFragment} = render(<MyListItem/>);
      expect(asFragment()).toMatchSnapshot();
    });
  });