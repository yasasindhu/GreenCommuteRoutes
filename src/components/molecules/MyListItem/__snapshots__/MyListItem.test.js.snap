// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendered listitemtextinlistitem 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-selectedItemStyle-1 MuiGrid-container MuiGrid-direction-xs-column"
    data-testid="divinmylistitem"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
      data-testid="listiteminmylistitem"
    >
      <div
        class="MuiListItemIcon-root"
        data-testid="listitemiconinlistitem"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-selectedIcon-2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root"
        data-testid="listitemtextinlistitem"
      >
        <h6
          class="MuiTypography-root makeStyles-selectedTypographyStyle-3 MuiTypography-subtitle1"
        >
          Saved Jobs
        </h6>
      </div>
    </li>
  </div>
</DocumentFragment>
`;
